sdks/java/extensions/sql/zetasql/udf-test-provider/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/provider/UdfTestProvider.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql.zetasql.provider;
+
+import com.google.auto.service.AutoService;
+import java.lang.reflect.Method;
+import java.util.Map;
+import org.apache.beam.sdk.extensions.sql.UdfProvider;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableMap;
+
+/**
+ * Defines Java UDFs that are tested in {@link
+ * org.apache.beam.sdk.extensions.sql.zetasql.ZetaSqlJavaUdfTest}.
+ */
+@AutoService(UdfProvider.class)
+public class UdfTestProvider implements UdfProvider {
+  @Override
+  public Map<String, Method> userDefinedScalarFunctions() {
+    try {
+      return ImmutableMap.of(
+          "fun",
+          this.getClass().getMethod("matches", String.class, String.class),
+          "foo",
+          this.getClass().getMethod("foo"),
+          "increment",
+          this.getClass().getMethod("increment", Long.class),
+          "isNull",
+          this.getClass().getMethod("isNull", String.class));
```

Comment:
As mentioned before, this is the place where I would like to see a test case like `"udfName", x -> x*2`